### PR TITLE
Added log error for config file not found

### PR DESCRIPTION
**Pull request by Christian-B**
Fixed one of many places where galaxy simply ignores a possible error.
Without any feedback to the admin.

### CODE REVIEW

File: lib/galaxy/tools/deps/__init__.py [hunk 1]
```diff
@@ -90,7 +90,10 @@ def find_dep( self, name, version=None, type='package', **kwds ):
         return INDETERMINATE_DEPENDENCY
 
     def __build_dependency_resolvers( self, conf_file ):
-        if not conf_file or not os.path.exists( conf_file ):
+        if not conf_file:
+            return self.__default_dependency_resolvers()
+        if not os.path.exists( conf_file ):
+            log.error( "Unable to find config file '%s'", conf_file)
```
Comment (nsoranzo) [inline lib/galaxy/tools/deps/__init__.py]: I'd use `log.debug()` here, this configuration file is not mandatory AFAIK.

**Comment by Christian-B** [comment]
Thank you @nsoranzo for the comment.

While new to galaxy project I look at this from the point of view of a galaxy admin.
I fully agree the config file is not mandatory so fully agree that if not specified nothing should be logged, or as suggested at most a debug.

Here however an admin has changed galaxy.ini to include a config file!
If a manually specified config file is not found that is an error.
True it is a administrator error and not a galaxy error but still an error.

**Comment by nsoranzo** [comment]
@Christian-B No, even with a vanilla `galaxy.ini` file

`galaxy.tools.deps ERROR 2015-07-09 13:41:47,685 Unable to find config file './dependency_resolvers_conf.xml'`

is added to the log file because `config.dependency_resolvers_config_file` defaults to `dependency_resolvers_conf.xml`, see https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config.py#L484 and following lines.

**Comment by Christian-B** [comment]
Thank you @nsoranzo for the clear explanation and link.

Sorry I was unaware of the default setting.
Not having set the default would have been nice but now clearly impossible due to reverse compatibility.

Fully agree that in this case a debug level is correct.

Pull requested changes as recommended.

**Comment by nsoranzo** [comment]
:+1: Thanks for the contribution!

**Comment by natefoo** [comment]
+1

**Pull request merged**